pkg/datastore/datastore_with_txcontext_test.go

```diff
@@ -65,58 +70,56 @@ func testTxCrud(t *testing.T, ds datastore.DataStore, ctx context.Context, myCok
 	user2.Email = "jahangir111@mail.com"
 	user2.EmailConfirmed = !user2.EmailConfirmed
 	user2.NumFollowers--
-
-	// TODO: Find, Update and Delete are still not supported due to appending where clauses,
-	// after each call to Find/Update/Delete methods
-
-	tx, err := ds.Helper().GetDBTransaction(ctx, datastore.GetTableName(user1), user1)
+	tx, err = ds.Helper().GetDBTransaction(ctx, datastore.GetTableName(user1), user1)
 	assert.NoError(err)
 	defer rollbackTx(t, tx)
-	txCtx := txFetcher.WithTransactionCtx(ctx, tx)
+	txCtx = txFetcher.WithTransactionCtx(ctx, tx)
 	for _, record := range []*AppUser{user1, user2} {
-		rowsAffected, err := ds.Upsert(txCtx, record)
+		rowsAffected, err := ds.Update(txCtx, record)
 		assert.NoError(err)
 		assert.EqualValues(1, rowsAffected)
-	}
-	assert.NoError(tx.Commit().Error)
 
-	for _, record := range []*AppUser{user1, user2} {
 		queryResult := &AppUser{Id: record.Id}
-		err = ds.Find(ctx, queryResult)
+		err = ds.Find(txCtx, queryResult)
 		assert.NoError(err)
 		assert.Equal(record, queryResult)
 	}
+	assert.NoError(tx.Commit().Error)
 
+	t.Log("Verifying upsert and find in single transaction")
+	user1.NumFollowers++
+	user2.NumFollowers--
 	tx, err = ds.Helper().GetDBTransaction(ctx, datastore.GetTableName(user1), user1)
 	assert.NoError(err)
 	defer rollbackTx(t, tx)
 	txCtx = txFetcher.WithTransactionCtx(ctx, tx)
-	// Upsert operation should be an update for already existing records
-	user1.NumFollowers++
-	user2.NumFollowers--
 	for _, record := range []*AppUser{user1, user2} {
 		rowsAffected, err := ds.Upsert(txCtx, record)
 		assert.NoError(err)
 		assert.EqualValues(1, rowsAffected)
-	}
-	assert.NoError(tx.Commit().Error)
-	for _, record := range []*AppUser{user1, user2} {
+
 		queryResult := &AppUser{Id: record.Id}
-		err = ds.Find(ctx, queryResult)
+		err = ds.Find(txCtx, queryResult)
 		assert.NoError(err)
 		assert.Equal(record, queryResult)
 	}
+	assert.NoError(tx.Commit().Error)
 
-	// Deletion of existing records should not fail, and the records should no longer be found in the DB
+	t.Log("Verifying deletion and find in single transaction")
+	tx, err = ds.Helper().GetDBTransaction(ctx, datastore.GetTableName(user1), user1)
```

Comment:
Same

Please use ds.GetTransaction() to init transaction. ds.Helper().GetDBTransaction() (or ds.GetDBTransaction()) is used in every CRUD function to either extract the transaction pointer out of ctx, or init a brand new one

Reply:
Done